src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -324,6 +326,18 @@ protected AnalyzerDriver(ImmutableArray<DiagnosticAnalyzer> analyzers, AnalyzerM
             return builder.ToImmutable();
         }
 
+        private ImmutableDictionary<DiagnosticAnalyzer, ImmutableArray<CompilationAnalyzerAction>> MakeCompilationActionsByAnalyzer()
+        {
+            var builder = ImmutableDictionary.CreateBuilder<DiagnosticAnalyzer, ImmutableArray<CompilationAnalyzerAction>>();
+            var actionsByAnalyzers = this.analyzerActions.CompilationActions.GroupBy(action => action.Analyzer);
```

Comment:
Maybe so. I copied this from existing code, and I've tried to completely minimize logic differences in this change set.

I'll merge the introduced method with the one above it (which I copied).